src/stitch.rs

```diff
@@ -85,25 +86,39 @@ impl Iterator for IterStitchedIndexHunks {
                         return Some(hunk);
                     } // otherwise, empty, try the next
                 }
-                if self.archive.band_is_closed(&self.band_id).unwrap_or(false) {
-                    return None;
-                }
+                /* We iterated through any know entry. */
                 self.index_hunks = None;
-                if let Some(band_id) = previous_existing_band(&self.archive, &self.band_id) {
-                    self.band_id = band_id;
-                } else {
+
+                let band_id = self.band_id.take().expect("last band id should be present");
```

Comment:
I realized the nice thing about `take` here is that if we return with opening a new band it leaves it as None.

Reply:
> I was trying to work out if this could ever panic.

Logically it can't panic. `index_hunks` can only be `Some` if `band_id` is `Some`.  
Using `take` results in `band_id` being `None` which in case of a closed band (complete index) means we're done iterating bands. If the current band has not been closed, `band_id` will be overridden by the earlier band which might be `None`. Using take here was an explicit decision.